rfcs/react-components/no-action-prop-value-standardization.md

```diff
@@ -0,0 +1,190 @@
+  // Required as we need a value for our hooks to work
+  appearance: 'filled' | 'filled-alternative' | 'outline' | 'subtle';
+};
+
+// useCard.ts
+const { appearance = 'filled' /* {...} */ } = props; // Default applied to enforce behavior
+
+const state = {
+  appearance,
+  // {...}
+};
+
+// useCardStyles.ts
+const appearanceLookup = {
+  filled: styles.filled,
+  'filled-alternative': styles.filledAlternative,
+  outline: styles.outline,
+  subtle: styles.subtle,
+} as const;
+
+state.root.className = mergeClasses(
+  cardClassNames.root,
+  styles.root,
+  appearanceLookup[state.appearance],
+  // {...}
+  state.root.className,
+);
+```
+
+Cases where nothing is done in our hooks as a default, we should leverage `undefined`.
+i.e. `Text` component's `font` prop should be nullable instead of using a `'default'` string value, as that reflects the fact that there are no side-effects from it.
+
+```tsx
+// Text.types.ts
+type TextProps = {
+  font?: 'base' | 'monospace' | 'numeric';
+};
+type TextState = {
+  // Also nullable as the default does not overwrite styles
+  font?: 'base' | 'monospace' | 'numeric';
+};
+
+// useText.ts
+const { font /* {...} */ } = props; // We no longer set the default here
+
+const state = {
+  font,
+  // {...}
+};
+
+// useTextStyles.ts
+const { font = '_default' } = state; // Applying a default to keep using a lookup
+
+const fontLookup = {
+  _default: undefined,
+  base: styles.fontBase,
+  monospace: styles.fontMonospace,
+  numeric: styles.fontNumeric,
+} as const;
+
+state.root.className = mergeClasses(
+  cardClassNames.root,
+  styles.root,
+  fontLookup[font],
+  // {...}
+  state.root.className,
+);
+```
+
+### Usage difference
+
+```ts
+// Before
+useTextStyles_unstable({
+  // User has to check docs to double check what the default shoud be.
+  align: 'start',
+  block: false,
+  font: 'monospace', // Only change from defaults
+  italic: false,
+  size: 300,
+  strikethrough: false,
+  truncate: false,
+  underline: false,
+  weight: 'regular',
+  wrap: true,
+});
+
+// After
+useTextStyles_unstable({
+  font: 'monospace',
+  // {...} - Required props
+});
+```
+
+## Pros and Cons
+
+### Pros
+
+- Follows JavaScript standards
+- Intuitive to the user because of the above pro - no need to read documentation
+- Consistency across our product - enables ease of use
+
+### Cons
+
+- Using the lookup object pattern for styling would require changes
+
+  Proposal:
+
+  <!-- prettier-ignore -->
+  ```tsx
+  const state = { align: undefined } as const;
+
+  const alignMap = {
+    _default: undefined,
+    start: styles.alignStart,
+    center: styles.alignCenter,
+    end: styles.alignEnd,
+    justify: styles.alignJustify,
+  } as const;
+
+  const className = mergeClasses(
+    classNames.root,
+    styles.root,
+    alignMap[props.align ?? '_default']
+  );
+  ```
+
+- Users who dynamically set a property value would need to do something like:
+  ```tsx
+  <Text align={isItFriday ? 'center' : undefined} />
```

Comment:
What is the 'before' version of this ? I'm not sure how this affects props

Reply:
The before would be something like:

```tsx
<Text align={isItFriday ? 'center' : 'none'} />
```

Dynamically setting the property will require an explicit passing of an `undefined` value.